### PR TITLE
Add more reduction types to nll_loss.

### LINKED ISSUE #1271: Change the "reduction" of Pytorch Losses
## 🚀 Feature
 Many torch loss functions, e.g., cross entropy, nll loss, etc., have an ability to change the `reduce` from mean to other values. This is used in multiple places inside Fairseq, where we would like to not take the mean and instead use the sum or not reduce the loss at all. This is especiallyuseful for changing the normalization / doing masking. For example, here https://github.com/pytorch/fairseq/blob/master/fairseq/criterions/masked_lm.py#L54

Currently, the loss functions are supported in XLA, but setting the reduce flags is unsupported and the objects move to the CPU (i.e., I see `aten:` calls for the cross entropy). This slows speeds significantly 

In the meantime, if your end goal is to have a `sum` instead of a `mean`, you can simply call the regular loss and then multiply by the number of elements. However, there are also cases where we need `none` reduction, which there is no work around that I am aware of.

### PR DESCRIPTION
**Pull request by dlibenzi**
Fixes #1271

### CODE REVIEW

File: test/cpp/test_aten_xla_tensor.cpp [hunk 1]
```diff
@@ -5958,33 +5958,45 @@ TEST_F(AtenXlaTensorTest, TestNllLoss) {
...
+                /*weight=*/xla_weight,
+                /*reduction=*/reduction, /*ignore_index=*/ignore_index);
+            AllClose(output, xla_output);
+          });
+        }
       }
     }
   }
+
+  ExpectCounterNotChanged("aten::(?!_local_scalar_dense).*",
```
Comment (ailzhang) [inline test/cpp/test_aten_xla_tensor.cpp]: Hmmm is part of test calling into `item()`?

File: torch_xla/csrc/aten_xla_bridge.cpp [hunk 2]
```diff
@@ -82,6 +82,10 @@ XLATensor GetOrCreateXlaTensor(const at::Tensor& tensor, const Device& device) {
   return xtensor ? *xtensor : XLATensor::Create(tensor, device);
 }
 
+XLATensor GetXlaTensorOrNull(const at::Tensor& tensor) {
```
Comment (ailzhang) [inline torch_xla/csrc/aten_xla_bridge.cpp]: This functions is very similar to `GetOrCreateXlaTensor`, is it intended?

Comment (dlibenzi) [reply torch_xla/csrc/aten_xla_bridge.cpp]: The difference is that if `tensor` is defined, it must be an XLA tensor.
But now that I think more about those curve ball coming from PT, with tensors from other devices being passed, we might just use the other one.

File: test/cpp/test_aten_xla_tensor.cpp (hunk 1 above)
Comment (dlibenzi) [reply test/cpp/test_aten_xla_tensor.cpp]: Hmm, IIRC it did. But maybe I am confusing it with the backward test.
Let me double check ...
